besu/src/main/java/org/hyperledger/besu/cli/util/CommandLineUtils.java

```diff
@@ -51,20 +51,59 @@ public static void checkOptionDependencies(
       final boolean isMainOptionCondition,
       final List<String> dependentOptionsNames) {
     if (isMainOptionCondition) {
-      final String affectedOptions =
-          commandLine.getCommandSpec().options().stream()
-              .filter(
-                  option ->
-                      Arrays.stream(option.names()).anyMatch(dependentOptionsNames::contains)
-                          && !option.stringValues().isEmpty())
-              .map(option -> option.names()[0])
-              .collect(
-                  Collectors.collectingAndThen(
-                      Collectors.toList(), StringUtils.joiningWithLastDelimiter(", ", " and ")));
+      final String affectedOptions = getAffectedOptions(commandLine, dependentOptionsNames);
 
       if (!affectedOptions.isEmpty()) {
         logger.warn(DEPENDENCY_WARNING_MSG, affectedOptions, mainOptionName);
       }
     }
   }
+
+  /**
+   * Check if options are passed that require an option to be true to have any effect and log a
+   * warning with the list of affected options. Multiple main options may be passed to check
+   * dependencies against.
+   *
+   * <p>Note that in future version of PicoCLI some options dependency mechanism may be implemented
+   * that could replace this. See https://github.com/remkop/picocli/issues/295
+   *
+   * @param logger the logger instance used to log the warning
+   * @param commandLine the command line containing the options we want to check
+   * @param mainOptions the names of the main options to test dependency against. Only used for
+   *     display.
+   * @param isMainOptionCondition the conditions to test the options dependencies, if true will test
+   *     if not won't
+   * @param dependentOptionsNames a list of option names that can't be used if condition is met.
+   *     Example: if --miner-coinbase is in the list and condition is that --miner-enabled should
+   *     not be false, we log a warning.
+   */
+  public static void checkMultiOptionDependencies(
+      final Logger logger,
+      final CommandLine commandLine,
+      final List<String> mainOptions,
+      final List<Boolean> isMainOptionCondition,
```

Comment:
I think it would be more intuitive to pass a single boolean expression rather than a list

Reply:
As in, pass a single parameter of `--privacy-enabled && --goquorum-compatibility-enabled` ?

Reply:
yes - they don't get used separately, and passing them in separately sort of makes it seem like they will be